### PR TITLE
109 filter sort

### LINKED ISSUE #109: Backend filter and sort routes
# Why
A public visitor of the archive may want to filter the archive by different attributes. 

# What to do

1. Create and test backend routes to get all anthologies in the database, and return sorted  by

- title (A-Z)
- pub date (most recent first)
- pub date (oldest first)

2. Create and test backend routes to return only anthologies with common 
- ageCategory
- pubDate (within range)
- program
- genre

# Acceptance Criteria

1. A user can retrieve all anthologies sorted by title (A-Z), author (A-Z), most recent publication date, and oldest publication date, each via a dedicated backend route.
2. A user can retrieve anthologies filtered by age category, publication date, program, and genre, each via a dedicated backend route.
3. Any filter / sort involving a string (rather than enum - so such as genre and program) should work regardless of case. So mystery, Mystery, and MYSTERY should be treated as the same genre.
4. All routes return only anthologies that exist in the database, return an empty list when no results match, and return correct HTTP status codes.
5. All routes have passing tests covering the happy path, empty results, and invalid inputs.

### PR DESCRIPTION
**Pull request by Ryaken-Nakamoto**
### ℹ️ Issue

Closes #109

### 📝 Description

Did a bit of bug bashing to run the backend. Not sure if it was running on your end, but after pulling from main I made the following changes:
1) Added Relation<> to prevent circular dependencies
2) Changed migrations to *.js from *.ts to enable node to target compiled output
3) added missing dependencies to various .module files

There was also a preexisting linting error for importing to the backend from the frontend. I disabled it for now with a // eslint* comment, unsure how you want to handle this.

EDIT AFTER PR REVIEW: Updated from 7 endpoints to 1 centralized endpoint to sort and filter. /filter-sort. Tested w/ no filters passed in, almost all of them passed in, and some in isolation. Also tested a case where an empty list was returned. Great callout from @GauriRajesh733 to change to this approach. Made all fields optional, thought it would have good ease of use for consumers of the service. Undefined/[] cases are handled in the service method. 

P.S. if you have trouble running the tests I had to regenerate my lock file, didn't commit it but if that is an issue on your end that is probably the solution.

### CODE REVIEW

File: apps/backend/src/anthology/dtos/filter-anthology.dto.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { IsString, IsNotEmpty, IsEnum, IsDateString } from 'class-validator';
+import { AgeCategory } from '../types';
+
+export class FilterByStringDto {
```
[the remark this answers is not in the document]
Comment (Ryaken-Nakamoto) [reply apps/backend/src/anthology/dtos/filter-anthology.dto.ts]: implemented, except for the sort DTO used an enum

File: apps/backend/src/anthology/anthology.controller.ts [hunk 2]
```diff
@@ -11,12 +13,23 @@ import { AnthologyService } from './anthology.service';
 import { Anthology } from './anthology.entity';
 import { AuthGuard } from '@nestjs/passport';
 import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+import { FilterSortAnthologyDto } from './dtos/filter-anthology.dto';
 
 @ApiTags('Anthologies')
 @Controller('anthologies')
 export class AnthologyController {
   constructor(private readonly anthologyService: AnthologyService) {}
```
[the remark this answers is not in the document]
Comment (Ryaken-Nakamoto) [reply apps/backend/src/anthology/anthology.controller.ts]: That's true, but passing in a body for GET requests is undefined in http specification, could we keep it with POST?
